roles/mysql/tasks/main.yml

```diff
@@ -0,0 +1,199 @@
+    src: my.cnf.j2
+    dest: /root/.my.cnf
+    mode: "0640"
+  when:
+    - mysql_server
+
+- name: MySQL | create wildcard user
+  community.mysql.mysql_user:
+    host: "{{ item }}"
+    name: root
+    password: "{{ mysql_root_password }}"
+    login_host: "{{ mysql_host }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  loop:
+    - "{{ ansible_hostname }}"
+    - 127.0.0.1
+    - ::1
+    - localhost
+  when:
+    - mysql_server
+    - running_on_server
+
+- name: MySQL | remove Anonymous users in the database
+  community.mysql.mysql_user:
+    name: ''
+    host: "{{ item }}"
+    state: absent
+  with_items:
+    - localhost
+    - "{{ ansible_hostname }}"
+    - 127.0.0.1
+  when:
+    - mysql_server
+    - running_on_server
+
+- name: MySQL | Remove the test database
+  community.mysql.mysql_db:
+    name: test
+    state: absent
+  when:
+    - mysql_server
+    - running_on_server
+  notify:
+    - restart mysql server
+
+- name: MySQL | create databases on server
+  community.mysql.mysql_db:
+    name: "{{ item.name }}"
+    state: "{{ item.state | default('present') }}"
+    target: "{{ item.target | default(omit) }}"
+    encoding: "{{ item.encoding | default(omit) }}"
+    collation: "{{ item.collation | default(omit) }}"
+    login_host: "{{ mysql_host }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  loop: "{{ mysql_databases }}"
+  loop_control:
+    label: "{{ item.name }}"
+  when:
+    - mysql_databases is defined
+    - mysql_server
+    - running_on_server
+  no_log: true
+
+- name: MySQL | create databases
+  community.mysql.mysql_db:
+    name: "{{ item.name }}"
+    state: "{{ item.state | default('present') }}"
+    target: "{{ item.target | default(omit) }}"
+    encoding: "{{ item.encoding | default(omit) }}"
+    collation: "{{ item.collation | default(omit) }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_host: "{{ mysql_host }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  delegate_to: "{{ mysql_host }}"
+  loop: "{{ mysql_databases }}"
+  when:
+    - mysql_databases is defined
+    - running_on_server
+
+- name: MySQL | create users on server
+  community.mysql.mysql_user:
+    name: "{{ item.name }}"
+    state: present
+    password: "{{ item.password | default(omit) }}"
+    priv: "{{ item.priv | default(omit) }}"  # least privilege access preferred
+    host: "{{ item.host | default(omit) }}"
+    host_all: "{{ item.host_all | default(omit) }}"
+    update_password: "{{ item.update_password | default(omit) }}"
+    login_host: "{{ mysql_host }}"
+    login_unix_socket: "{{ mysql_socket }}"
+    login_user: root
+    login_password: "{{ mysql_root_password }}"
+  loop: "{{ mysql_users }}"
+  when:
+    - mysql_users is defined
+    - mysql_server
+    - running_on_server
+  no_log: true
+
+- name: MySQL | create users
```

Comment:
We have two scenarios - when we're installing a server, and when we're not. Should this task run in both scenarios? Either way, a better `name:` could clarify (`create the same users on the client` or `create client users` or something). If it should only run when we're not installing a server, then add a condition of `not mysql_server`.